src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx

```diff
@@ -141,6 +162,21 @@ describe('useSidebarMetadataFetcher', () => {
         expect(result.current.errorMessage).toBeNull();
     });
 
+    test('should fetch metadata and return only visible templates and template instances', async () => {
+        const { result } = setupHook();
+
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({
+                templates: [...mockTemplates, mockHiddenTemplate],
```

Comment:
Can we also have tests for only hidden template instances?

Reply:
Change the approach and move the logic to `MetadataSidebarRedesign` and added tests to corresponding file